crates/rune-macros/src/parse.rs

```diff
@@ -95,28 +95,60 @@ impl Expander {
         input: &syn::DeriveInput,
         named: &syn::FieldsNamed,
     ) -> Option<TokenStream> {
+        let ast_attributes_ident: syn::Ident = syn::parse2(quote! {attributes})
+            .expect("Parse proc_macro could not create needed resource Ident(\"attributes\")");
+
+        let mut has_ast_attributes = false;
         let mut fields = Vec::new();
 
         for field in &named.named {
             let _ = self.ctx.parse_field_attributes(&field.attrs)?;
             let ident = self.ctx.field_ident(&field)?;
+            if ident.eq(&ast_attributes_ident) {
```

Comment:
Nice. I like that you're generating `parse_with_attributes` functions since they all follow the same pattern. Hopefully we can do more of that!

Could you make this optional behavior activated by an attribute `#[rune(attributes)]` instead? I prefer the explicitness of it over defaulting to the name. You can see [how they are parsed here](https://github.com/rune-rs/rune/blob/master/crates/rune-macros/src/context.rs#L134).

As for synthesizing identifiers (the parse/quote thing), check out the [`internals` module](https://github.com/rune-rs/rune/blob/master/crates/rune-macros/src/internals.rs#L8). A `Symbol` can be used for that (it implements `ToTokens` for this purpose).

Reply:
Can do. After making the "parse -> parse_with_attributes"  pattern few times I thought I'd give generating it a go.